blocks/api/factory.js

```diff
@@ -45,9 +45,15 @@ export function switchToBlockType( block, blockType ) {
 		return null;
 	}
 
+	const attributes = transformation.transform( block.attributes );
+	if ( attributes instanceof Error ) {
+		// Blocks can perform validation and cancel transformations if needed.
+		return attributes;
```

Comment:
Returning `null` above is also an error (no transformation found), we should be consistent. 

In the same time, I'm wondering if we really need to throw specific errors. I don't see any use case where we'd want to perform different actions depending on the error. Should we show this error somewhere in the UI? I don't think so, it's not relevant for the user I think.

Note that the `transforms` API is not only used for block switching, if we introduce new returns values like these, we'd want to handle them in all "clients" see #457 

Also, I think we should perform the transformation even if we have data loss (we should limit the data loss) because the user explicitly asks for the transformation, so it's not a surprise for him to get its content changed a bit. For example when switching from a text block to a heading block, I'm not agains concatening with spaces the different paragraphs. If we reject the transformation, this means backspacing from a text block to a heading won't work either.